mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -157,6 +166,12 @@ private DB extractDatabase(MongoClient mongo, MongoDBConfiguration config) {
 					throw log.databaseDoesNotExistException( config.getDatabaseName() );
 				}
 			}
+			DB db = mongo.getDB( databaseName );
+			if ( containsDatabase == null ) {
+				// force a connection to make sure we do have read access
+				// otherwise the connection failure happens during the first flush
+				db.collectionExists( "WeDoNotCareWhatItIsWeNeedToConnect" );
```

Comment:
There might be but I haven't found one. There is not explicit API nor a `select 1 = 1` like query either that I could find.